### DISCUSSION TITLE
How to enable Vue.js dev tool to inspect the web extension under development?

### DISCUSSION THREAD

**Discussion by albertpratomo** [antfu/vitesse-webext#150] [Q&A] [answered]
<img width="222" alt="image" src="https://github.com/antfu/vitesse-webext/assets/25815584/a7822f4c-948f-4f50-b258-9521ff5290c7">

I would love to develop my web ext using Vue.js dev tool, but it's not detecting it now.. Any idea?

**Comment by albertpratomo** [comment, marked as answer]
I realize the app is actually served in my localhost:
```
http://localhost:3303/background/index.html
http://localhost:3303/options/index.html
http://localhost:3303/popup/index.html
```

So I can go there and see the app, also use the Vue dev tool.

However, by default it will throw error because `webextension-polyfill` can't be used outside of Chrome context.

To work around that, instead of always using browser storage, we can use `localStorage` as fallback.

I will make a PR for this soon.

**Reply by jay-p-b-7span** [reply]
Hey @albertpratomo , Could you please share further steps to start using vue.js devtools ? If Needed, I would be happy to contribute

**Reply by CodyBontecou** [reply]
I run into this console error when navigating to any of these urls.

![CleanShot 2024-07-16 at 21 49 16@2x](https://github.com/user-attachments/assets/48934166-055e-4ae1-b010-afc50e91de93)
